optuna/terminator/erroreval.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+import abc
+
+import numpy as np
+
+from optuna._experimental import experimental_class
+from optuna.study.study import Study
+from optuna.trial import Trial
+from optuna.trial._state import TrialState
+
+
+_CROSS_VALIDATION_SCORES_KEY = "terminator:cv_scores"
+
+
+class BaseErrorEvaluator(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def evaluate(self, study: Study) -> float:
+        pass
+
+
+@experimental_class("3.2.0")
+class CrossValidationErrorEvaluator(BaseErrorEvaluator):
+    def evaluate(self, study: Study) -> float:
+        assert len(study.get_trials(states=(TrialState.COMPLETE,))) != 0
+
+        best_trial_attrs = study.best_trial.system_attrs
+        if _CROSS_VALIDATION_SCORES_KEY in best_trial_attrs:
+            cv_scores = best_trial_attrs[_CROSS_VALIDATION_SCORES_KEY]
+        else:
+            raise ValueError(
+                "Cross-validation scores have not been reported. Please call "
+                "`report_cross_validation_scores(trial, scores)` during a trial and pass the "
+                "list of scores as `scores`."
+            )
+
+        k = len(cv_scores)
+        scale = 1 / k + 1 / (k - 1)
+
+        var = scale * np.var(cv_scores)
+        std = np.sqrt(var)
+
+        return float(std)
+
+
+@experimental_class("3.2.0")
+def report_cross_validation_scores(trial: Trial, scores: list[float]) -> None:
```

Comment:
Sorry for not having mentioned this sooner but I'm finding it just slightly confusing to have this function under the same sub-module as `StaticErrorEvalator` . I think we can leave it as is though as those are experimental APIs. (We could make it a static method of `CrossValidationErrorEvaluator` but it's a matter of preference).

Reply:
To make the API call short and simple, let us keep `report_cross_validation_scorers` an isolated function.